tests/PrestashopWebServiceTest.php

```diff
@@ -1,18 +1,138 @@
 <?php
 
-use Illuminate\Foundation\Testing\WithoutMiddleware;
-use Illuminate\Foundation\Testing\DatabaseMigrations;
-use Illuminate\Foundation\Testing\DatabaseTransactions;
+namespace Protechstudio\PrestashopWebService\Tests;
+
 use Protechstudio\PrestashopWebService\PrestashopWebService;
+use Protechstudio\PrestashopWebService\Exceptions\PrestashopWebServiceException;
+use Protechstudio\PrestashopWebService\Exceptions\PrestashopWebServiceRequestException;
+use Protechstudio\PrestashopWebService\PrestashopWebServiceLibrary;
+use Prestashop;
 
 class PrestashopWebServiceTest extends TestCase
 {
-
-    /**
-     * @test
-     */
+    /** @test */
     public function it_is_correctly_installed()
     {
         $this->assertInstanceOf(PrestashopWebService::class, Prestashop::getFacadeRoot());
     }
+    
+    /** @test */
+    public function it_can_perform_a_get_request()
+    {
+        $requestResponseStub = require('requests/category-schema.php');
+        $ps = $this->getMockedLibrary('executeCurl', $requestResponseStub);
```

Comment:
`$ps = $this->getMockedLibrary('executeCurl', $requestResponseStub);` is correct? Shouldn't it be `$ps = $this->getMockedLibrary('executeRequest', $requestResponseStub);`

Reply:
1. Changing it to `executeRequest` the error changes:
```
There was 1 error:

1) Protechstudio\PrestashopWebService\Tests\PrestashopWebServiceTest::it_can_perform_a_get_request
ErrorException: Undefined index: status_code

/home/vagrant/code/laravel-prestashop-webservice/src/PrestashopWebServiceLibrary.php:78
/home/vagrant/code/laravel-prestashop-webservice/src/PrestashopWebServiceLibrary.php:369
/home/vagrant/code/laravel-prestashop-webservice/tests/PrestashopWebServiceTest.php:25

ERRORS!
Tests: 8, Assertions: 14, Errors: 1.
```
2. As far as I know this depends from system as Windows is case-insensitive while linux is case-sensitive but I'll conduct a test to see if I'm correct.

Reply:
No, mocking `executeCurl` is correct in this test. Part of the test is whether `executeRequest` handles a curl response correctly. Thats the reason I separate the curl logic to a separate method, this way we can easily test/mock the curl stuff to easily test our logic that handles that curl response. 
What you can to check if the short open tag is the problem is to add `ini_set('short_open_tags', 0);` in the test method. Or change `<?xml` to `'<' . '?xml'.

If you really want to be sure, you could maybe add travis (linux) / appveyor (windows) configuration for automatic build tests for php5.4, php7.0. Its the easiest to check things across different systems. Just copy and adapt the config from other projects :)

Reply:
I'm actually testing on Vagrant homestead with php7.1 and checked that short_open_tags is off as it should be so it's not the problem. There's something that is strange anyway, if you read the error (first one) it declares `'file' => '',`.
I was already thinking about adding CI with travis but didn't have time yet. I'll take a look now

Reply:
Could it maybe be an issue with linebreaks, so `\n` vs `\r\n`? See the new test I added